src/components/structures/RoomView.js

```diff
@@ -782,32 +792,51 @@ export default createReactClass({
                     e2eStatus: hasUnverifiedDevices ? "warning" : "verified",
                 });
             });
+            debuglog("e2e check is warning/verified only as cross-signing is off");
             return;
         }
+
+        /* At this point, the user has encryption on and cross-signing on */
         const e2eMembers = await room.getEncryptionTargetMembers();
-        for (const member of e2eMembers) {
-            const { userId } = member;
-            const userVerified = cli.checkUserTrust(userId).isCrossSigningVerified();
-            if (!userVerified) {
-                this.setState({
-                    e2eStatus: "warning",
-                });
-                return;
-            }
+        const verified = [];
+        const unverified = [];
+        e2eMembers.map(({userId}) => userId)
+            .filter((userId) => userId !== cli.getUserId())
+            .forEach((userId) => {
+                (cli.checkUserTrust(userId).isCrossSigningVerified() ?
+                verified : unverified).push(userId)
+            });
+
+        debuglog("e2e verified", verified, "unverified", unverified);
+
+        /* If we verify any users in this room, expect to verify every user in the room */
```

Comment:
Sorry, this still isn't quite right: a room should only be `warning` if there exists a verified user that has an unverified device (ie. a user with a red shield). What you can say here is `if (verified.length === 0) this.setState({e2eStatus: "normal"});}`

Reply:
I think that's covered by iterating over an empty list, no need to special-case

Reply:
oh I see, you meant a differnt bit of code – sorry, just reading the code itself I realise what you had tagged here